pusher/photoservice.go

```diff
@@ -43,6 +45,11 @@ func (r *remoteAPI) existingPhotos() ([]string, error) {
 			return []string{}, err
 		}
 
+		if len(r.username) > 0 || len(r.password) > 0 {
+			// HTTP basic auth
+			req.SetBasicAuth(r.username, r.password)
```

Comment:
Cleartext.... Base64 is not a secure hash...

Reply:
So, you're suggesting something like Digest Auth? https://en.wikipedia.org/wiki/Digest_access_authentication

Reply:
Digest utilizes MD5 which is not secure. Most authentication libraries wrap basic-auth by hashing the password clientside

Reply:
Generally speaking I'd suggest using an authentication library, like oauth1

Reply:
https://github.com/abbot/go-http-auth or even this is fine

Reply:
We should store and compare in hash, not in cleartext

Reply:
Second thought, I less care if trasport is hashed, we should have https for that. But storage should be definitely be hashed

Reply:
Oh, I guess someone could determine the length of the username or password because the [constant time comparisons](https://github.com/kochman/hotshots/blob/59126a6df3b42232bb10a9d50b412996afa38880/server/auth.go#L19) return instantly if the lengths don't match.

Reply:
On reflection, you should instead provide a hash as the env-var, and the server hashes input. Our app shouldn't know your password, just the hash of your password, and compare the input after it's been hashed